python/ml4ir/applications/ranking/tests/test_auxiliary_loss.py

```diff
@@ -0,0 +1,128 @@
+import gc
+import os
+import pathlib
+import unittest
+import warnings
+
+import numpy as np
+import pandas as pd
+import tensorflow.keras.backend as K
+from testfixtures import TempDirectory
+
+from ml4ir.applications.ranking.config.parse_args import get_args
+from ml4ir.applications.ranking.pipeline import RankingPipeline
+
+warnings.filterwarnings("ignore")
+
+ROOT_DATA_DIR = "ml4ir/applications/ranking/tests/data"
+
+
+def train_ml4ir(data_dir, feature_config, model_config, logs_dir, aux_loss):
```

Comment:
@mohazahran FYI - old test
https://github.com/salesforce/ml4ir/blob/1f4ef603f132949c2c88a3ed852733fcf9de95f3/python/ml4ir/applications/ranking/tests/test_auxiliary_loss.py

Two changes that might be impacting the results (I've verified this by simply changing some of it and running the old tests successfully).
- Changing `y_true` -> `tf.multiply(y_true, mask)`
- Masked softmax